sql: TestCreateStatsAfterSchemaChange failed

sql.TestCreateStatsAfterSchemaChange [failed](https://tanzanite.cluster.engflow.com/invocations/default/b801cc73-460d-4704-9f2a-e615be2e7273?testReportRun=1&testReportShard=15&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [bd4c5fd7c65c0a9afa8ba05675e49747a4f68881](https://github.com/cockroachdb/cockroach/commits/bd4c5fd7c65c0a9afa8ba05675e49747a4f68881):


```
=== RUN   TestCreateStatsAfterSchemaChange
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCreateStatsAfterSchemaChange1651073927
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    sql_runner.go:121: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:369
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/schema_changer_test.go:5100
        	Error:      	Received unexpected error:
        	            	query 'SELECT statistics_name, column_names, row_count, distinct_count, null_count
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:372
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:75
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:81
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:369
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestCreateStatsAfterSchemaChange
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/schema_changer_test.go:5100
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) query 'SELECT statistics_name, column_names, row_count, distinct_count, null_count
        	            	  | 	  FROM [SHOW STATISTICS FOR TABLE t.test] ORDER BY column_names::STRING': expected:
        	            	  | __auto__, {k}, 0, 0, 0
        	            	  | __auto__, {v}, 0, 0, 0
        	            	  | __auto__, {w}, 0, 0, 0
        	            	  |
        	            	  | got:
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestCreateStatsAfterSchemaChange1651073927
--- FAIL: TestCreateStatsAfterSchemaChange (55.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=15</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsAfterSchemaChange.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39438

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -5071,7 +5071,8 @@ CREATE TABLE t.test (a INT, b INT, c JSON, d JSON);
 func TestCreateStatsAfterSchemaChange(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.UnderRace(t, "sometimes the timer in stats.Refresher doesn't fire fast enough under race")
+
+	skip.UnderDuress(t, "sometimes the timer in stats.Refresher doesn't fire fast enough under custom configs")
 
 	defer func(oldRefreshInterval, oldAsOf time.Duration) {
 		stats.DefaultRefreshInterval = oldRefreshInterval

